### ISSUE TITLE
Subchips / Consumable items

### ISSUE THREAD

**Issue by ArthurCose** [ArthurCose/Scriptable-OpenNetBattle-Server#16]
May need a similar API to "key items" / items, to provide description. Rather than working as a simple list we also need subchip counts.

I'd like to see the name change away from Subchip to something like consumable. It might make sense to consider a consumable as an item to avoid duplicate API (recycling name + description functions and packets).

Minimally we'll need this:
```lua
Net.set_player_consumable_count(player_id, item_id, count)
Net.get_player_consumable_count(player_id, item_id)
```

**Comment by TheMaverickProgrammer** [comment]
Would be great to re-use the item API. I do think we will need to identify them differently:

`Net.create_item(item_id, { name, description, type })` where `type=KEY_ITEM` reserved value is a key item (default) and `type=CONSUMABLE` is the new one.

We can let the client tally up the counts. So if a player has 3 `ElectricSpray` in their inventory, the client will receive `['ElectricSpray', 'ElectricSpray', 'ElectricSpray']` and display `ElectricSpray x3`.

**Comment by ArthurCose** [comment]
I'd be worried about high count consumables, such as a max of 50 or 99. The games had a lower max count of 4 so it might not be an issue if server owners stick closer to the games. This also brings up a `max` count we might want to add to `Net.create_item`.

I think we should also have a way to give multiple of the same item at once so giving a player 5 of some item does not require a loop. (maybe an optional count param to `Net.give_item` / `Net.remove_item`)

**Comment by TheMaverickProgrammer** [comment]
I understand these points but it means refactoring for ezlibs and some rewriting on the client. What if we keep the internals the same but change the way we send that data to the client? So if a user does have 99x items represented as a big array, the client will only be sent a short packet of `[{"ElectricSpray", 3}]`.

**Comment by ArthurCose** [comment]
This might be slow when a script makes a call to get the player's inventory. (copying 99 strings through Net.get_player_items)

**Comment by TheMaverickProgrammer** [comment]
Yeah. It would be better to supply counts for everything item related. I guess we just need to get ezlibs to support this change.
